prime-router/docs/proposals/0006-email-notification/0006-email-notification.md

```diff
@@ -0,0 +1,174 @@
+
+The email service will store its relevant data within the Postgres database, with the exception of templates which will be maintained on the SendGrid site (with references to them in Postgres).
+
+The EmailService will take in a Schedule element that consists of a body similar to:
+``` json
+{
+    "template": "template.name", // the name of the template
+    "schedule": "chron string for scheduling", // optional, no schedule means send immediate
+    "organization" : "optional organization to send to", // ex. DHpima_az_doh, DHmt_doh, etc.
+    "recipient": [ "optional", "optional" ], //optional array of recipients emails
+    "excluding": [ "optional"], // a list of recipients to exclude, otherwise everybody within the organization receives
+    "parameters": {
+        "param1": "value for param1",
+        ...
+    }
+}
+```
+
+See [openapi.emailservice.yml](./openapi.emailservice.yml) for a detailed specification
+
+The EmailService will fire off emails based on:
+- If there are organizations, only for those organizations, otherwise, for every organization
+- If there are recipients, only for those recipients, otherwise, for every recipient, with the exception of anyone listed in "excluding"
+
+The EmailService will record all of its actions to a logging table.
+
+The EmailService will require an HttpHeader Authentication with a valid JWT token.
+
+### Special parameter values
+
+| Parameter | Description |
+| --------  | ------------|
+|$reports_since_last| retrieves an array of reports sent since the last email |
+|$today | Java Date representing the current date/time |
+|$org | The organization being referenced |
+|$user | The username being referenced |
+
+*Note: more to be added*
+
+## Examples
+
+### List all active EmailSchedules
+#### Request
+```http
+GET /email-service/schedule HTTP/1.1
+```
+#### Response
+```json
+HTTP/1.1 200 OK
+
+[
+    {
+        "id": "2130-9991",
+        "template": "daily.tmpl",
+        "schedule": "0 0 0 12", 
+        "parameters": {
+            "reports": "$reports_since_last",
+            "today": "$today",
+            "organization": "$org"
+        }
+    },
+    {
+        "id": "2130-9991",
+        "template": "weekly.tmpl",
+        "schedule": "0 0 0 12", 
+        "organization": [ "DHpima_az_doh" ],
+        "except": ["joe.jones@az.pima.gov"],
+        "parameters": {
+            "reports": "$reports_since_last",
+            "organization": "$org"
+        }
+    }
+]
+```
+#### Notes
+In this case, we have 2 email schedules, the first uses the template "daily.tmpl" and is scheduled to fire daily at noon, with paramters for reports, today, and organization.  The second, uses a template "weekly.tmpl" and goes to everyone in the pima_az_doh organization (except for joe.jones) once a week on friday at noon.
+
+### Create an EmailSchedule
+#### Request
+```json
+POST /email-service/schedule HTTP/1.1
+
```

Comment:
I think of the schedule as just another setting. EG. /settings/email-schedule. This way we don't have write another CLI, Facade, etc. Instead just extend the current one.